### PR TITLE
[Redesign] Update Qdtrack operator

### PR DESCRIPTION
**Pull request by objorkman**
* Update `QdTrackOperator` to redesign

### CODE REVIEW

**Review by pschafhalter — APPROVED** [review]
LGTM. Let's wait until #240 is merged, so we can fix the conflicts and make sure it passes CI.

**Comment by sukritkalra** [comment]
I've merged #240 so @objorkman if you can rebase and fix the conflicts, we can run the CI once again and pull this in too.

**Pull request merged**
